src/include/network/postgres/statement.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "common/managed_pointer.h"
+#include "network/postgres/statement.h"
+#include "parser/postgresparser.h"
+#include "traffic_cop/traffic_cop_util.h"
+#include "type/type_id.h"
+
+namespace terrier::network {
+
+/**
+ * Statement is a postgres concept (see the Extended Query documentation:
+ * https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-EXT-QUERY)
+ * It encapsulates a parsed statement, the parameter types (if any). It represents a statement ready to be turned into a
+ * Portal with a Bind message.
+ */
+class Statement {
```

Comment:
Maybe we can name it more specifically, like postgres_statement? Even though this is in the networking layer, I think it would be better to differentiate this from general statements.